src/display.rs

```diff
@@ -147,6 +201,29 @@ impl Display for Table {
     }
 }
 
+impl Document {
+    /// Returns a string representation of the TOML document, attempting to keep
+    /// the table headers in their original order.
+    ///
+    /// The best case performance of this function is slightly slower than
+    /// Document.to_string(). If you have lots of tables that are in strange
+    /// orders then it may be significantly slower as it has to walk the tree
+    /// multiple times.
+    pub fn to_string_in_original_order(&self) -> std::result::Result<String, Error> {
```

Comment:
The biggest open question is whether I should make the return signature for this method be like Display::to_string() by calling .unwrap() or something.

If you have suggestions for better names for anything then please say. If you think it would be an acceptable change for to_string() to implement this algorithm then we don't need to worry about naming or return types ;-) .

Reply:
I don't really understand conditions under which it can return an error. AFAIK, `write!` to string always succeeds, unless we are out of memory, in which case we can't do much about it.

Reply:
Okay. I will probably unwrap the error then.

Might not be until the weekend though. Thanks for the advice.